packages/gatsby/src/redux/reducers/pages.ts

```diff
@@ -1,31 +1,44 @@
-const normalize = require(`normalize-path`)
+import normalize from "normalize-path"
+import { ActionsUnion, IGatsbyPage, IGatsbyState } from "../types"
 
-module.exports = (state = new Map(), action) => {
+export const pagesReducer = (
+  state: IGatsbyState["pages"] = new Map<string, IGatsbyPage>(),
+  action: ActionsUnion
+): IGatsbyState["pages"] => {
   switch (action.type) {
     case `DELETE_CACHE`:
       return new Map()
+
     case `CREATE_PAGE`: {
       action.payload.component = normalize(action.payload.component)
-      if (!action.plugin && !action.plugin.name) {
+
+      // throws an error if the page is not created by a plugin
+      if (!action.plugin || !action.plugin.name) {
```

Comment:
```suggestion
      if (!action.plugin?.id) {
```
If you change this to a check on `id` then you can lose the fallback empty strings on lines 27-28

Reply:
That was my first instinct too @ascorbic. But I was not really sure it was the intent of this check. If it's the case, I'm happy to update this check to remove the fallback on empty strings :+1:

Reply:
I wasn't sure either, but checking where plugin is created, it generates the id based on the name anyway